analyze_data.py

```diff
@@ -115,7 +130,11 @@ def get_all_data_as_dict(filename):
         dict: Dictionary with all data from parse_filename and
               get_data_from_file
     """
-    return {}
+    filedata = parse_filename(filename)
+    data = get_data_from_file(filename)
+    data_dictionary = filedata.copy()
+    data_dictionary.update(data)
+    return {data_dictionary}
```

Comment:
What is the type of data_dictionary?  

You are attempting to return a set with only one item, a dictionary.  Instead we just need to return a dictionary